tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -208,6 +210,178 @@ def setUp(self):
+        Create mock data for a dual-written alert rule.
+        Return all models that we expect for a dual written alert rule.
+        """
+        detector_data_condition_group = self.create_data_condition_group(
+            organization=metric_alert.organization
+        )
+        query_subscription = QuerySubscription.objects.get(snuba_query=metric_alert.snuba_query)
+        data_source = self.create_data_source(
+            organization=metric_alert.organization,
+            query_id=query_subscription.id,
+            type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION,
+        )
+        detector = self.create_detector(
+            project_id=self.project.id,
+            enabled=True,
+            created_by_id=self.rpc_user.id if self.rpc_user else None,
+            name=metric_alert.name,
+            workflow_condition_group=detector_data_condition_group,
+            type=MetricAlertFire.slug,
+            description=None,
+            owner_user_id=None,
+            owner_team=None,
+            config={
+                "threshold_period": 1,
+                "sensitivity": None,
+                "seasonality": None,
+                "comparison_delta": None,
+                "detection_type": AlertRuleDetectionType.STATIC,
+            },
+        )
+        workflow = self.create_workflow(
+            name=metric_alert.name,
+            organization_id=metric_alert.organization.id,
+            when_condition_group=None,
+            enabled=True,
+            created_by_id=self.rpc_user.id if self.rpc_user else None,
+            config={},
+        )
+        data_source.detectors.set([detector])
+        detector_state = self.create_detector_state(
+            detector=detector,
+            active=False,
+            state=DetectorPriorityLevel.OK,
+        )
+        alert_rule_detector = AlertRuleDetector.objects.create(
+            alert_rule=metric_alert, detector=detector
+        )
+        alert_rule_workflow = AlertRuleWorkflow.objects.create(
+            alert_rule=metric_alert, workflow=workflow
+        )
+        detector_workflow = DetectorWorkflow.objects.create(detector=detector, workflow=workflow)
+
+        return (
+            data_source,
+            detector_data_condition_group,
+            workflow,
+            detector,
+            detector_state,
+            alert_rule_detector,
+            alert_rule_workflow,
+            detector_workflow,
+        )
+
+    def mock_migrate_alert_rule_trigger(
+        self, alert_rule_trigger
+    ) -> tuple[DataCondition, DataCondition]:
+        """
+        Create mock data for a dual-written alert rule trigger. Migrate the alert rule first, as required.
+        Return all models that we expect for a dual written alert rule trigger.
+        """
+        alert_rule = alert_rule_trigger.alert_rule
+        aci_tables = self.mock_migrate_alert_rule(alert_rule)
+        detector_data_condition_group = aci_tables[1]
+        workflow = aci_tables[2]
```

Comment:
Could you look up the models you need rather than calling `mock_migrate_alert_rule` and just make sure you call that before you call this function?